tests/test_psnr.py

```diff
@@ -16,49 +17,45 @@ def target() -> torch.Tensor:
 
 
 # ================== Test function: `psnr` ==================
-def test_psnr_works_for_colour_images(prediction: torch.Tensor, target: torch.Tensor) -> None:
-    psnr(prediction, target, data_range=1.0)
-
-
-@pytest.mark.skipif(not torch.cuda.is_available(), reason='No need to run test if there is no GPU.')
-def test_psnr_works_for_colour_images_on_gpu(prediction: torch.Tensor, target: torch.Tensor) -> None:
-    prediction = prediction.cuda()
-    target = target.cuda()
-    psnr(prediction, target, data_range=1.0)
-
-
-def test_psnr_works_for_greyscale_images() -> None:
-    prediction = torch.rand(4, 1, 256, 256)
-    target = torch.rand(4, 1, 256, 256)
-    psnr(prediction, target, data_range=1.0)
-
-
-@pytest.mark.skipif(not torch.cuda.is_available(), reason='No need to run test if there is no GPU.')
-def test_psnr_works_for_greyscale_images_on_gpu() -> None:
-    prediction = torch.rand(4, 1, 256, 256).cuda()
-    target = torch.rand(4, 1, 256, 256).cuda()
-    psnr(prediction, target, data_range=1.0)
-
-
+def test_psnr(input_tensors: Tuple[torch.Tensor, torch.Tensor], device: str) -> None:
+    prediction, target = input_tensors
+    psnr(prediction.to(device), target.to(device), data_range=1.0)
+
+
+@pytest.mark.parametrize(
+    "data_range", [128, 255],
```

Comment:
General remark to all the series of tests: could you add more options of the `data_range` values if it is not too computationally heavy? It would be very interesting to see what happens if `data_range` is 1.0, <1.0 (for instance 0.5) of some weird huge number like 3000.